### ISSUE TITLE
"whatprovides" feature - to find what package would provide a certain file

### ISSUE THREAD

**Issue by ZvikaZ** [spack/spack#9694]
Add a new feature, "whatprovides", that will show what package provides a desired file.

### Rationale

Suppose I would like to install some file (e.g. "make"), but there isn't a package with that name.
How would I know what package should I install in order to get that file?

### Description

This can be solved by a "whatprovides" feature, similar to one exists in "yum", or other package managers.

**Comment by michaelkuhn** [comment]
I agree that this would be a useful feature. It will probably require built binary packages to be available (#7123) because Spack currently does not know which files are installed by a given package.

**Comment by healther** [comment]
And this could quickly escalate. There are packages with a rather large set of variants and version around... Should this only return binaries? Or also libraries? headers?
